security/advancedtls/testdata/server_cert_localhost_1.pem

```diff
@@ -59,66 +59,67 @@ Certificate:
             X509v3 Key Usage: 
                 Digital Signature, Key Encipherment
             X509v3 Subject Alternative Name: 
-                DNS:localhost, IP Address:127.0.0.1
+                DNS:localhost, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
```

Comment:
@dfawley @menghanl 
Pulling you both in since we had a discussion about this during the import process.
Are you Ok with this change?

Reply:
It's probably also worth mentioning the reason why certificate with only "localhost" as the SAN name doesn't work. IIRC, it worked on my local machine, but while this was tested on the presubmit tests, it failed, and I noticed that the authority name(https://github.com/grpc/grpc-go/blob/master/security/advancedtls/advancedtls.go#L380) fell back to the ip address, while at the time when we did the connection, we connected to a "localhost"(https://github.com/grpc/grpc-go/blob/master/security/advancedtls/advancedtls_test.go#L694).
That was the original reason why I wanted to also add ip address to the certificates. But just to double check: is this behavior expected?

Reply:
Your test is doing a direct `net.Dial` and calling `ClientHandshake` itself.

https://github.com/grpc/grpc-go/blob/c638ab8ccda6cefc15eb9a25357fc98b018ede37/security/advancedtls/advancedtls_test.go#L760

I believe the listener address is expected to become the IP address.  Maybe you should call it with "localhost" instead since you're invoking it directly.

Is there a reason these tests are invoking these things directly instead of going through `grpc.Dial` / `grpc.Server` (how users would use them)?  Are there more e2e-style tests that use grpc?

Reply:
Ah I see. Yeah, I put all the tests using `grpc.Dial` in `advancedtls_integration_test.go`. These tests were in `advancedtls_test.go` before because I thought they were simple behavior so I just treated them as unit tests. But now I  think it's better to put them in integration tests, and use `grpc.Dial` / `grpc.Server`. I've made such changes. Could you please take a look? Thank you so much!

Reply:
Can you delete the IP address entries from here now?

Reply:
Seems no, because It's still checking against the ip address...I was connecting to `localhost:0`, but from the logs, the authority string in `ClientHandshake(ctx context.Context, authority string, rawConn net.Conn)` was automatically converted to `127.0.0.1:XXXXX`(XXXXX is a random port number). I attached the logs I used to debug [here](https://gist.github.com/ZhenLian/9f8490acaa75b4af9d159550f312702b). I am wondering if this is normal?

Reply:
I don't understand.  This is how *you* are calling `ClientHandshake` in your tests.  Don't use the address from the listener; use "localhost", and it should work without the IP addresses.